walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java

```diff
@@ -22,6 +22,25 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
-    return "Hello " + name;
+    return "Hello " + removeSymbols(name.trim());
+  }
+  
+  private String removeSymbols(String name) {
+    String newString = name;
+    for (int i = 0; i < name.length(); i++) {
+      char character = name.charAt(i);
+      String charSequence = "" + character + "";
+      int ascii = (int) character;
+      //Allowing for A-Z, a-z, ', and -.
+      if (!((ascii > 64 && ascii < 91) || (ascii > 96 && ascii < 123) || ascii == 39 || ascii == 45)) {
+        System.out.println("Made it into if statement with: " + charSequence);
+        newString = replaceCharacter(newString, charSequence);
+      }
+    }
```

Comment:
I think this can be replaced with a single call to replaceAll() (https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replaceAll(java.lang.String,%20java.lang.String) ) Note that this method accepts a regex, which you may not be familiar with yet. The regex `[^a-zA-Z]` will match all characters that are not an ASCII letter, and you can replace them with `""`. So the call would be `replaceAll("[^a-zA-Z]", "");`

Reply:
Note that if you want to remove only leading and trailer whitespace, you can call `trim()` on the string first, then use the regex `[^a-zA-Z ]`. This is also probably a case you should add a test for.